packages/dds/tree/src/tableSchema.ts

```diff
@@ -96,7 +118,42 @@ export namespace TableSchema {
 			IColumn<TPropsSchema> &
 			WithType<ScopedSchemaName<Scope, "Column">>;
 
-		type ColumnInsertableType = InsertableObjectFromSchemaRecord<typeof columnFields>;
+		// Note: ideally this type would just leverage `InsertableObjectFromSchemaRecord<typeof columnFields>`,
+		// but that results in broken `.d.ts` output due to a TypeScript bug.
+		// Instead we extract and inline the typing of the "props" field here, which seems to sufficiently work around the issue.
```

Comment:
It would be good to link the relevant TypeScript bug here.

Reply:
Or maybe reference the docs on RowInsertableType and dedupliocate these comments and link it there? SOmehow we should centralize the documentation about this issue, perhaps even centralizing it in the TypeScript issue or a comment about our specific case on it.